### ISSUE TITLE
Symlinks are not updated after a reboot

### ISSUE THREAD

**Issue by brahman81** [oogali/ebs-automatic-nvme-mapping#2]
Hi Oogali,

Thanks for researching and documenting this C5/M5 instance family nvme issue we are all plagued with!

While implementing your solution, I noticed that the current script does not handle incorrect symlinks. By this I mean that on an instance with multiple EBS volumes already attached and a new volume is added  it is possible that when this instance is rebooted it comes back with different mapping than on previous runs of the script, this is especially the case with multiple EBS volumes attached.

For what it's worth, in our limited observations the smaller EBS volumes appear to have higher precedence than larger EBS volumes.

To remedy this I have written a new bash script which is heavily inspired by your research and wanted to submit it to you. The script is available on my gist and if you feel it worthwhile I am happy to submit a PR.

https://gist.github.com/brahman81/04d1d19b81f78385aacf0a9a3a3afe26

Kind Regards,

**Comment by brahman81** [comment]
fwiw, I have updated the script as we were having issues with udev, script only returns mapping information, symlinks are now created by udev.

**Comment by oogali** [comment]
Thanks! Can you submit a PR for review?

**Comment by zymergen-vdc** [comment]
@brahman81 you're not actually creating the symlink in your gist, so I'm also very curious to see your PR. thanks!!

**Comment by brahman81** [comment]
@vdczzz, the original version of the script (version 1) did create the symlinks but what we found was that symlinks would only get created during initial attachment of the volumes which didn't help with reboot,  we would see the symlinks but they were not mounted as udev decided to unmount immediately post mount.

The latest version of the script doesn't actually create any symlinks, but instead given a block device as `$1` it returns the correct mapping name and lets udev create/manage the symlink instead. (There is a udev rule to be added in the comments of version 2)

Basically we are passing the return of the udev `PROGRAM` key to the udev `SYMLINK` key.

`SUBSYSTEM=="block", KERNEL=="nvme[0-9]*n[0-9]*", ATTRS{model}=="Amazon Elastic Block Store", PROGRAM+="/usr/local/sbin/nvme-mapping.sh /dev/%k" SYMLINK+="%c"`

This is more akin to what Amazon do in their AMI to deal with the problem.

**Comment by sssyam** [comment]
Also, I checked one more thing in the rule that there is no "remove" action.
When detaching the volume, the symlink is still persisting.

There was one more test case that I tried,
I detached the two volume one by one which were already attached to the instance and mounted them in reverse manner, the symlink was not changed.

Can you look at this pls ? 

Awesome work btw !!
